src/src/Commands/MQTT.cpp

```diff
@@ -74,4 +74,27 @@ String Command_MQTT_Publish(struct EventStruct *event, const char *Line)
   return return_not_connected();
 }
 
+//MFD: adding subscribe command
+String Command_MQTT_Subscribe(struct EventStruct *event, const char* Line)
+{
+  if (wifiStatus == ESPEASY_WIFI_SERVICES_INITIALIZED ) {
+    // ToDo TD-er: Not sure about this function, but at least it sends to an existing MQTTclient
```

Comment:
Hmmm that looks like cut-and-paste code ;)

Reply:
use `if (WiFiConnected()) {`

Reply:
from the publish command. the extern is because of another method that is defined in an .ino file. I am looking to get it moved now if possible inside the mqtt.cpp file

Reply:
The approach is taken (cut/paste/modified) from the publish command.
I am not sure about how we check that the MQTT client is connected thus I left you comment in.

Are you ok with this or should I investigate further ?
I am using this command for a year now inside the MQTT#connected event with no issues. but if somebody try to use it before connecting not sure of the effect.

Let me know what you think, please.

Reply:
> WiFiConnected

notted.

Reply:
https://github.com/letscontrolit/ESPEasy/blob/5707fb09a482c03406ab7cbf5cb73977bbe35108/src/src/Globals/MQTT.h#L16

Reply:
It seems I can't avoid the extern even moving that function inside the mqtt.cpp still references void setIntervalTimerOverride(unsigned long id, unsigned long msecFromNow); function from .ino file.

Reply:
can't find where it is definied `MQTTsubscribe`, did you commited this function?

Reply:
> from the publish command. the extern is because of another method that is defined in an .ino file. I am looking to get it moved now if possible inside the mqtt.cpp file

That's a very bad reason to use extern.
Anything in a .ino file is "converted" into a .cpp file at build. All .h includes are copy-paste into the same generated .cpp file and all .cpp files found during compile will be converted into object files.
By declaring something extern, you make it accessible from outside the object file. So if you have a conflict between functions, you leave it to the linker to decide what to link. So by forcing it extern you may break something else.

Reply:
> The approach is taken (cut/paste/modified) from the publish command.
> I am not sure about how we check that the MQTT client is connected thus I left you comment in.

It was rather funny to see my own comment in a commit of someone else :)